src/vmcalgs.inl

```diff
@@ -104,6 +105,72 @@ Positions<D, N> AddTo_(Positions<D, N> const &poss, Dimension d, ParticNum n, Co
     return result;
 }
 
+// Should be hidden from the user
+// Computes the drift force by numerically estimating the derivative of the wavefunction
+template <Dimension D, ParticNum N, VarParNum V, class Wavefunction>
+void DriftForceNumeric_(Wavefunction const &psi, VarParams<V> params, FPType derivativeStep,
+                        Positions<D, N> poss, Position<D> &driftForce) {
+    static_assert(IsWavefunction<D, N, V, Wavefunction>());
+    for (ParticNum n = 0u; n != N; ++n) {
+        for (Dimension d = 0u; d != D; ++d) {
+            driftForce[d] = 2 *
+                            (psi(PerturbPos_<D, N>(poss, d, n, Coordinate{derivativeStep}), params) -
+                             psi(PerturbPos_<D, N>(poss, d, n, Coordinate{-derivativeStep}), params)) /
+                            (derivativeStep * psi(poss, params));
+        }
+    }
+}
+
+// Updates the wavefunction with the Importance Sampling algorithms and outputs the number of succesful
+// updates
+template <Dimension D, ParticNum N, VarParNum V, class Wavefunction>
+UIntType ImportanceSamplingUpdate_(Wavefunction const &psi, VarParams<V> params, Positions<D, N> &poss,
+                                   FPType step, FPType deltaT, RandomGenerator &gen) {
+    static_assert(IsWavefunction<D, N, V, Wavefunction>());
+
+    UIntType successfulUpdates;
+    for (Position<D> p : poss) {
+        Position<D> const oldPos = p;
+        FPType const oldPsi = psi(poss, params);
+        Position<D> driftForce;
+        DriftForceAnalytic_(psi, poss, params, driftForce);
+        std::normal_distribution<FPType> normalDist(0.0, 1.0); // Gaussian noise
```

Comment:
Do not initialize with 0.0, 1.0, they are doubles.
Use floats, they will possibly be promoted.
The comment "Gaussian noise" is unnecessary.